cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -868,6 +884,15 @@ ColumnChunkMetaData const& aggregate_reader_metadata::get_column_metadata(size_t
                                                                           size_type src_idx,
                                                                           int schema_idx) const
 {
+  // schema_idx_maps will only have > 0 size when we are reading matching column projection from
+  // mismatched Parquet sources.
+  if (src_idx and schema_idx_maps.size()) {
+    auto const& schema_idx_map = schema_idx_maps[src_idx - 1];
+    CUDF_EXPECTS(schema_idx_map.find(schema_idx) != schema_idx_map.end(),
+                 "Found no metadata for the mapped schema index");
```

Comment:
```suggestion
    CUDF_EXPECTS(schema_idx_map.find(schema_idx) != schema_idx_map.end(),
                 "Found no metadata for the mapped schema index", std::invalid_argument);
```

@mhaseeb123 I'm suggesting this kind of change so that these errors are not runtime errors to users. We don't want to show runtime errors to user. They can be manifested as value errors, key errors, attribute errors, overflow errors etc...

Reply:
Makes sense, should I also change the behavior for the same check for all columns [here](https://github.com/rapidsai/cudf/blob/branch-24.10/cpp/src/io/parquet/reader_impl_helpers.cpp#L560) and [here](https://github.com/rapidsai/cudf/blob/branch-24.10/cpp/src/io/parquet/reader_impl_helpers.cpp#L557) before this PR for consistency? This PR simply skips these checks for later on in `map_column` function with more descriptive error string

Reply:
Updated. No `RuntimeError`s in Python anymore.